app/models/Notification.scala

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models
+import java.time.LocalDateTime
+
+import play.api.libs.json.Json
+
+case class Notification(dateTimeReceived: LocalDateTime, conversationId: String) extends Ordered[Notification] {
```

Comment:
How come we're using `LocalDateTime`?

If we run this service in two timezones we would get two different LocalDateTimes for what is essentially the same time. It doesn't carry any timezone info it would falsely appear in the DB as being different.

We should use `Instant` which is a UTC date

Reply:
Here is the same approach like in exports frontend. We just copied this.

But the question is good. I think we can change LocalDateTime, but we need to do it in every microservice. Not only here. I feel new ticket for it.

Reply:
@ed0906 `LocalDateTime` does not carry any timezone, so that would not be an issue.
`ZonedDateTime` keeps a timezone and in that instance we would see a different time if we kept the value in the DB as UTC (zulu time).

Am I correct?

Reply:
https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/time/LocalDateTime.html
`
This class does not store or represent a time-zone. Instead, it is a description of the date, as used for birthdays, combined with the local time as seen on a wall clock. It cannot represent an instant on the time-line without additional information such as an offset or time-zone.
`

Reply:
So, as far as I understand, as long as we capture the client's local time correctly and store it in the DB and present it back it is fine _**as long as the client does not change timezone**_ (not the server).
If submit a declaration in the UK today at 13h and then view it in Germany the following day I should see today's date and 14h (if we kept the zone information _Instant/UTC/ZonedDateTime_), but If I saw the time 13h it would be OKish.
We should use ZonedDateTime and store the date/time in an epock/instant format in the DB.
But I think this is a user need requirement, although heavily influenced by the technological implementation, but this requires a further discussion with BAs to understand what should a customer see on our FE? A simple LocalDateTime or a ZonedDateTime?

Reply:
@PauloPortugal It's exactly because LocalDateTime doesn't contain timezone info that *is* the issue. We either want to use UTC time (`Instant`) or some date type *with* a timezone (`ZonedDateTime`). Given `Instant` is generally easier to use given the average operation we tend to do (e.g. saving it in a DB or converting to a String) I think the general consensus is to use that

Reply:
> @ed0906 `LocalDateTime` does not carry any timezone, so that would not be an issue.
> `ZonedDateTime` keeps a timezone and in that instance we would see a different time if we kept the value in the DB as UTC (zulu time).
> 
> Am I correct?

The JSON serialization for Mongo would convert it to a Mongo Date Type which is basically a mirror image of what  an `Instant` is (number of milliseconds since 1/1/1970 in UTC time). As a result ZonedDateTime would be fine as the serialization process would know how to get UTC time from it. 
With LocalDateTimes how do you convert that to UTC? You have to *presume* the time zone which is why we shouldnt use them.

Reply:
Guys, as I understand LocalDateTime will be date from the server, not specific user, so we don't have problem with this. It's not necessary to capture zone etc.

Reply:
From documentation: `Obtains the current date-time from the system clock in the default time-zone.`

Reply:
I have changed to ZonedDateTime as it carries information about the timezone.
@MaciejRewera please adjust the DB to store an instant/epoch